### ISSUE TITLE
"Apns Delegate Never Fired" while using Firebase Auth with and without swizzling

### ISSUE THREAD

**Issue by moonvader** [OneSignal/OneSignal-Flutter-SDK#507]
**Description:**
I have issue with OneSignal push notifications for iOS app after I added Firebase Auth with phone. All iOS devices listed in OneSignal are marked with "Apns Delegate Never Fired" label and notifications can be sended but app don't receive them.

I found two similar issues

- https://github.com/OneSignal/OneSignal-iOS-SDK/issues/790
- https://github.com/OneSignal/OneSignal-Flutter-SDK/issues/245

As I understand the problem might be with swizzling that is used by Firebase Auth. I disabled swizzling using info.plist file as it suggested in the [official firebase docs](https://firebase.google.com/docs/auth/ios/phone-auth#appendix:-using-phone-sign-in-without-swizzling). But issue remains, so I think that I am missing something and OneSignal SDK still not being called.

My AppDelegate file looks like

```
override func application(_ application: UIApplication, didRegisterForRemoteNotificationsWithDeviceToken deviceToken: Data) {
           Auth.auth().setAPNSToken(deviceToken, type: AuthAPNSTokenType.unknown)
 }

 override func application(_ application: UIApplication, didReceiveRemoteNotification userInfo: [AnyHashable : Any], fetchCompletionHandler completionHandler: @escaping (UIBackgroundFetchResult) -> Void) {

           if Auth.auth().canHandleNotification(userInfo) {
               completionHandler(.noData)
               return
           }
}
```

**Environment**
onesignal_flutter: 3.2.4
flutter 2.5.3
Xcode 13.0

**Comment by jkasten2** [comment]
@moonvader Thanks for reporting. Can you attempt to temporally remove Firebase Auth from you project to confirm that the issue is related to any other changes you made to your project at the same time you added Firebase?

The only two things that can cause the "Apns Delegate Never Fired" status is 1. If you device can't connect to Apple's ANPS server to get a push token. This can happen if you're network block ports for example. 2. If there is another SDK / library that swizzles in an incompatible way.

Could you add logging to you `didRegisterForRemoteNotificationsWithDeviceToken` to see if it is firing? This will rule out the network issue.

Lastly could you note the firebase version you are using? Also if you have a example project reproducing the issue this will help us find the root cause.

**Comment by moonvader** [comment]
I am sure that this issues is connected with Firebase Auth problem started right after adding Firebase Auth with phone

firebase_auth: 3.1.4 pub is used

**Comment by jkasten2** [comment]
@moonvader Thanks for letting us know, we will attempt to reproduce the issue and report back.

**Comment by baraghiffari** [comment]
I have a 2 apps that uses onesignal, one that has this issue has a bunch of firebase plugin installed including firebase_auth 3.3.0 and the other one only firebase_core. My flutter version is 2.2.3

**Comment by rupas385** [comment]
@moonvader @jkasten2 Hi! I am fighting with the same problem, any update about it?

**Comment by adaskaliuk** [comment]
Any updates? Now we begin to think to say Bay Bay to OneSignal.

**Comment by jkasten2** [comment]
This issue was most likely fixed in the [onesignal_flutter 3.3.2 release](https://github.com/OneSignal/OneSignal-Flutter-SDK/releases/tag/3.3.2). Swizzling as been reworked in this version and should be more compatible with other SDKs.

If you are still seeing an issue please provide details on the exact versions of all SDKs you are using along with your Flutter version.

**Comment by abdelrahmanbonna** [comment]
@jkasten2 I am still having the same issue in the latest release
